### PR TITLE
Settings GUI: Hide advanced settings by default

### PR DESCRIPTION
**Pull request by rubenwardy**
Part of  #13476

Adds a new checkbox to show/hide advanced settings, hidden by default. By hiding advanced settings, the settings GUI is less overwhelming.

image

## To do

This PR is Ready for Review.

Questions:

- Resolve the ambiguity between different types of advanced settings (there's both an "Advanced" section and the advanced checkbox now)
- Finalise which settings are considered advanced

## How to test

* Run MT, open settings
* Toggle show advanced settings dialog, look at settings
* See all settings marked as advanced in settingtypes.txt

### CODE REVIEW

**Comment by SmallJoker** [comment]
The PR works but I think it would be good to have an indicator which categories that are "advanced". This is especially important for instructing players (wiki, chat). Suggestion: darker background of active and inactive states.

File: builtin/settingtypes.txt [hunk 1]
```diff
@@ -933,112 +1093,174 @@ movement_gravity (Gravity) float 9.81
...
 #    Small-scale humidity variation for blending biomes on borders.
+#
+#    Requires: show_advanced
 mg_biome_np_humidity_blend (Humidity blend noise) noise_params_2d 0, 1.5, (8, 8, 8), 90003, 2, 1.0, 2.0, eased
 
 [*Mapgen V5]
 
 #    Map generation attributes specific to Mapgen v5.
+#
+#    Requires: show_advanced
```
Comment (SmallJoker) [inline builtin/settingtypes.txt:1158]: This is very repetitive, especially for mapgens. Could this functionality be added for entire categories which is then propagated to contained settings?

**Comment by grorp** [comment]
> Resolve the ambiguity between different types of advanced settings (there's both an "Advanced" section and the advanced checkbox now)

I think this ambiguity is quite confusing. I would prefer it if all advanced settings were moved to the "Advanced" and "Mapgen" categories. Toggling the "Show advanced settings" checkbox would then only toggle the visibility of these two top-level categories.

**Comment by rubenwardy** [comment]
I'll make a new PR that hides page (IE: mapgen and advanced) rather than individual settings

**Pull request closed without merging**